src/main/java/com/jcabi/log/Logger.java

```diff
@@ -429,6 +429,32 @@ public void write(final int data) throws IOException {
     }
 
     /**
+     * Log messages constructed from Suppliers (Java 8 users).
```

Comment:
@amihaiemil what about such Java 7 code? It's bit verbose, but someone may accept it
```java
class Complex implements Supplier {
  void compute() {
    Logger.withSupplier().debug(this, "Some %s", this)
  }
  public String get() {
    return expensivePrint();
  }
}